### PR TITLE
[Improve][Paimon] Add check for the base type between source and sink before write.

### PR DESCRIPTION
**Pull request by dailai**
### Purpose of this pull request

Add check for the base type between source and sink before write.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java [hunk 1]
```diff
@@ -62,7 +62,15 @@ public enum CommonErrorCode implements SeaTunnelErrorCode {
             "COMMON-28",
             "'<identifier>' array type not support genericType '<genericType>' of '<fieldName>'"),
     UNSUPPORTED_ROW_KIND(
-            "COMMON-29", "'<identifier>' table '<tableId>' not support rowKind  '<rowKind>'");
+            "COMMON-29", "'<identifier>' table '<tableId>' not support rowKind  '<rowKind>'"),
+
+    WRITE_SEATUNNEL_ROW_ERROR_WITH_SCHEMA_INCOMPATIBLE_SCHEMA(
+            "COMMON-30",
+            "<connector>: The source filed with sql schema '<sourceFieldSqlSchema>', except sql schema of sink is '<exceptFieldSqlSchema>'; but the filed in sink table which actual sql schema is '<sinkFieldSqlSchema>'. Please check schema of sink table."),
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java]: Not all connectors use sql.
```suggestion
            "<connector>: The source filed with schema '<sourceFieldSqlSchema>', except schema of sink is '<exceptFieldSqlSchema>'; but the filed in sink table which actual schema is '<sinkFieldSqlSchema>'. Please check schema of sink table."),
```

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonError.java [hunk 2]
```diff
@@ -245,4 +247,28 @@ public static SeaTunnelRuntimeException unsupportedRowKind(
...
+        Map<String, String> params = new HashMap<>();
+        params.put("connector", connector);
+        params.put("sourceFieldSqlSchema", sourceFieldSqlSchema);
+        params.put("exceptFieldSqlSchema", exceptFieldSqlSchema);
+        params.put("sinkFieldSqlSchema", sinkFieldSqlSchema);
+        return new SeaTunnelRuntimeException(
+                WRITE_SEATUNNEL_ROW_ERROR_WITH_SCHEMA_INCOMPATIBLE_SCHEMA, params);
+    }
+
+    public static SeaTunnelRuntimeException writeRowErrorWithFiledsNotMatch(
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonError.java]:
```suggestion
    public static SeaTunnelRuntimeException writeRowErrorWithFiledsCountNotMatch(
```

File: seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/RowConverter.java [hunk 3]
```diff
@@ -462,4 +470,24 @@ public static InternalRow reconvert(
...
+    private static void checkCanWriteWithType(
+            int i, SeaTunnelRowType seaTunnelRowType, List<DataField> fields) {
+        String sourceFieldName = seaTunnelRowType.getFieldName(i);
+        SeaTunnelDataType<?> sourceFieldType = seaTunnelRowType.getFieldType(i);
+        DataField sinkDataField = fields.get(i);
+        DataType exceptDataType =
+                RowTypeConverter.reconvert(sourceFieldName, seaTunnelRowType.getFieldType(i));
+        DataField exceptDataField = new DataField(i, sourceFieldName, exceptDataType);
+        DataType sinkDataType = sinkDataField.type();
+        // Check typeRoot
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/utils/RowConverter.java]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-paimon-e2e/src/test/java/org/apache/seatunnel/e2e/connector/paimon/PaimonIT.java [hunk 4]
```diff
@@ -35,6 +36,7 @@
 
 @DisabledOnContainer(
         value = TestContainerId.FLINK_1_13,
+        type = {EngineType.FLINK, EngineType.SPARK},
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-paimon-e2e/src/test/java/org/apache/seatunnel/e2e/connector/paimon/PaimonIT.java]: Any reason for change this?

**Review by Hisoka-X — COMMENTED** [review]
Thanks @dailai for this PR!

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-paimon-e2e/src/test/java/org/apache/seatunnel/e2e/connector/paimon/PaimonIT.java (hunk 4 above)
Comment (dailai) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-paimon-e2e/src/test/java/org/apache/seatunnel/e2e/connector/paimon/PaimonIT.java]: Sorry, It's a mistake.

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java (hunk 1 above)
Comment (dailai) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java]: Done

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @dailai

**Review by hailin0 — APPROVED** [review]

**Pull request merged**